packages/react/src/components/form/flow.ts

```diff
@@ -27,6 +32,8 @@ export interface AuthenticatingState {
   cancel: Cancel;
   recover: () => void;
   entry: () => void;
+  hasUIState: (state: AuthenticatingUIStatus) => boolean;
```

Comment:
Using this naming convention is limiting us a bit, since the nested state does not have to be a UI state (eg it could be an ephemeral state that does some side effect).

Maybe doing something similar to [XState.matches](https://xstate.js.org/docs/guides/states.html#state-methods-and-properties) could help here? So you can pass the parent state name or the child state name, it should return the correct result anyway?